### PR TITLE
Break down analyzeQuery for more generation flexibility

### PR DESCRIPTION
**Pull request by CarsonF**
I'm trying to get immutable output types, without duplicating a bunch of this code.

### CODE REVIEW

File: packages/driver/src/reflection/index.ts [hunk 1]
```diff
@@ -30,4 +30,4 @@ export * from "./reservedKeywords";
 // export * from "../syntax/funcops";
 // export * from "./syntax";
 export * as introspect from "./queries";
-export * from "./analyzeQuery";
+export { analyzeQuery } from "./analyzeQuery";
```
Comment (scotttrinh) [inline packages/driver/src/reflection/index.ts:33]: Was this change necessary or are you being defensive here to avoid leaking the new exports?

**Review by scotttrinh — APPROVED** [review]
Looks good!

File: packages/driver/src/reflection/index.ts (hunk 1 above)
Comment (CarsonF) [reply packages/driver/src/reflection/index.ts:33]: Defensive. But I forgot until now that reflection/index isn't exported at top level. So many exporting these functions just to reflection is ok. Thoughts?

Comment (scotttrinh) [reply packages/driver/src/reflection/index.ts:33]: I don't have a strong opinion on this, but I guess it's safest to be restrictive now and add more to the export if we want them in the future, so good to keep what you have here.

Comment (CarsonF) [reply packages/driver/src/reflection/index.ts]: Ok I adjusted names to better fit a more generalized export/import.

The more I look at this the more I wonder why this is in the driver package and not the generation one...

**Review by scotttrinh — APPROVED** [review]
Dig it. 👍
